src/main/java/de/dennisguse/opentracks/util/PreferencesUtils.java

```diff
@@ -318,4 +345,76 @@ public static void setDefaultExportDirectoryUri(SharedPreferences sharedPreferen
     public static boolean isDefaultExportDirectoryUri(SharedPreferences sharedPreferences, Context context) {
         return getDefaultExportDirectoryUri(sharedPreferences, context) != null;
     }
+
+    public static int getLayoutColumns(SharedPreferences sharedPreferences, Context context) {
+        return getInt(sharedPreferences, context, R.string.stats_custom_layout_columns_key, context.getResources().getInteger(R.integer.stats_custom_layout_columns_default));
+    }
+
+    public static void setLayoutColumns(SharedPreferences sharedPreferences, Context context, int columns) {
+        setInt(sharedPreferences, context, R.string.stats_custom_layout_columns_key, columns);
+    }
+
+    private static List<TypedArray> getMultiTypedArray(Context context, String key) {
+        List<TypedArray> typedArrays = new ArrayList<>();
+
+        try {
+            Class<R.array> resource = R.array.class;
+            Field field;
+            int counter = 0;
+
+            do {
+                field = resource.getField(key + "_" + counter);
```

Comment:
Not a fan of manually referencing XML entries: this will prevent us using static code analyzer to find unused resources.

Reply:
I did it like this so developers can add more fields through settings.xml without the needed of update anything else.

So, if you want to add a new file you only have to add this new array to the settings.xml resources:

```xml
<string-array name="stats_custom_layout_fields_default_value_18">
        <item>@string/stats_custom_layout_newfield_key</item>
        <item>0,0,0</item>
    </string-array>
```

Appending the next number.

With this approach we cannot to use a constant resource's name.

Reply:
Works for me: still the parsing code and the generating for this string should be rather close (if there is a generating code).
And use a constant for: "_"